### ISSUE TITLE
ignore paths specified in `.babelrc` are not being respected when files are later created during watch mode `babel -w`

### ISSUE THREAD

**Issue by jeffbski** [babel/babel#1362]
ignore paths specified in `.babelrc` are not being respected when files are later created during watch mode `babel -w`

For instance if I have this in my `.babelrc`

```
{
  "ignore": [
    "**/flycheck*"
  ]
}
```

`babel -w` will ignore any flycheck* files on startup, but then **if a file is created later, it incorrectly runs on the file.**

However if I specify the ignore as a command line option `babel -w --ignore '**/flycheck*'` then everything works properly. It appears to only be an issue when trying to use `.babelrc`

**Comment by zcei** [comment]
Hej,

you might want to have a look at my PR to [enhance the docs](https://github.com/babel/babel.github.io/pull/257).

I guess, that the `.babelrc` is loaded like the command-line option at startup, but is further on treated by `api/register/node.js` which currently does not accept an array of globs.
The resulting RegExp would be `\\*\\*\\/flycheck\\*`(`_.escapeRegExp`) and thus not matching your needs

**Comment by jeffbski** [comment]
@zcei thanks for mentioning that, I believe you are correct that this is the likely cause of the problem

**Comment by zcei** [comment]
If you want to, you can try out to reference my fork in your `package.json` [zcei/babel](https://github.com/zcei/babel), where I added an `arrayify` wrapper as in the cli.

If it fixes this, I'll do a PR, as all tests are still running, and it's likely, that babel wants to have a unified usage.

**Comment by jeffbski** [comment]
@zcei have you pushed your change to your master branch? I don't see the commits on github.

**Comment by zcei** [comment]
My bad, forgot to commit after all the `make` foo before I could run the tests.
Should be there now

**Comment by jeffbski** [comment]
@zcei at first the change didn't show up since the compiled lib files weren't updated in your fork, but I manually applied the change to the lib code and unfortunately that didn't fix the issue for me. Thanks for trying.

**Comment by jeffbski** [comment]
@zcei I also had tried your work around by `{ "ignore": "**/flycheck*" }` and didn't seem to work either. Maybe there is something else messing things up.

**Comment by zcei** [comment]
@jeffbski Updated, so that it tests all arrays entries. 
If you pull & rebuild the core it might work.

Shortly after my answer I saw, that it doesn't handle an array where it tests against the regex.
Added a fix for this.

That it doesn't work with a single entry is sad, that should've done it :/ 
Now that I've put work into this.. let's see if I get it running. :D

**Comment by jeffbski** [comment]
@zcei I pulled your fork, rebuilt and linked to it replacing the babel-core in my node_modules, but it is still not ignoring created files during watch. :-(

**Comment by zcei** [comment]
Could you provide me an minimal example where I can observe it?
Tried to watch a folder containing a non-matching file & a matching, and then dropping in a matching while it's running.

But for me it does just copy over the file and replaces the extension.

**Comment by jeffbski** [comment]
@zcei I have created a repo here which can reproduce the issue https://github.com/jeffbski/babel-ignore see the README.md for detailed instructions.

**Comment by jeffbski** [comment]
@zcei If I understand how ignore works, it shouldn't do anything with ignored files. So it shouldn't be copying the file or anything, so destination directory should not have the ignored file.

**Comment by zcei** [comment]
Ah okay, I reference you to [the docs](http://babeljs.io/docs/usage/options/).

It does copy over your files, but doesn't compile it.

So modifying your second command to 
```
echo "var odds = [1,3,5].map(v => v + 1);" > src/flycheck_foo.js
```
should result in an unmodified version in `build-lib`

**Comment by jeffbski** [comment]
@zcei I don't see in the docs where it indicates anywhere that ignored files should be copied over. It seems like it should simply ignore them always. Like it does when specified on command line.

**Comment by zcei** [comment]
@jeffbski I agree, that it should ignore completely, but there should also be an option to just copy over, as it breaks the paths otherwise.

Docs for `only`:
> When attempting to compile a non-matching file it's returned verbatim.

For `ignore`:
> Opposite to the only option.

**Comment by sebmck** [comment]
@zcei That's not the same as `$ babel` `--ignore`.

**Comment by jeffbski** [comment]
ok. I see now. Yes, there should be a way to ignore completely and if necessary a way to copy other files.

**Comment by zcei** [comment]
@sebmck I had a second look, and if I understand it correct now, there is an `--ignore` for skipping the require hook and one for the actual CLI to ignore files?

I see the difference, but could you clarify the intended behavior of the `ignore` relevant to this case?

**Comment by sebmck** [comment]
The `ignore` option, does just that, it ignores the input. For the CLI, it just doesn't copy the file, for `babel/register` and `babel.transform` it just doesn't do any transforms on the file. The problem here is that `.babelrc` options aren't being coerced in the CLI.

**Comment by zcei** [comment]
This should be unified, if you ask me.

In my interpretation the docs say, that both the CLI and `babel.transform` options work the same way.

**Comment by sebmck** [comment]
It's different because the CLI and the node API function very differently. You provide a directory to the Babel CLI, it can decide what files it compiles. The node API cannot just throw away files because the consumer is expecting a response.

**Comment by jeffbski** [comment]
That makes sense since if I am using transform in something like browserify with babelify I can exclude files from the input of browserify. Using babel and babel -w ignore means don't include at all (which is just what I want).

**Comment by sebmck** [comment]
Fixed. Extremely sorry for the delay, will be available in the next patch due out shortly.

**Comment by sebmck** [comment]
Fixed as of 5.4.0, thanks!

**Comment by jeffbski** [comment]
Thanks!
